src/js/common/parser.js

```diff
@@ -0,0 +1,281 @@
+    var requireDefinitions = that.resolveJsonPointer(schemaKey, failurePointer);
+    var requirementIndex = fluid.find(requireDefinitions, function (value, index) {
+        if (value === propertyToMatch) { return index; }
+    });
+
+    // The path to the message is relative to the parent
+    if (requirementIndex !== undefined && requirementIndex !== null) {
+        var parentPointer = gpii.schema.parser.getParentJsonPointer(failurePointer);
+        var errorsPointer = gpii.schema.parser.getChildJsonPointer(parentPointer, "required");
+        return gpii.schema.parser.getChildJsonPointer(errorsPointer, requirementIndex);
+    }
+    else {
+        return undefined;
+    }
+};
+
+/**
+ *
+ * A static function to return the last segment of a given JSON pointer.
+ *
+ * @param jsonPointer {String} - The original JSON pointer.
+ * @returns A {String} representing the last segment of the original pointer.
+ */
+gpii.schema.parser.getLastJsonPointerSegment = function (jsonPointer) {
+    var segments = jsonPointer.split("/");
+    return segments[segments.length - 1];
+};
+
+/**
+ *
+ * A model listener to dereference and cache a dereferenced version of schemas as they are added to `model.schemas`.
+ */
+gpii.schema.parser.updateSchemas = function (that) {
+    var promises = [];
+    fluid.each(
+        that.model.schemas, function (schemaContent, schemaKey) {
+        if (!that.dereferencedSchemas[schemaKey]) {
+            promises.push(that.dereferenceSchema(schemaKey));
+        }
+    });
+
+    if (promises.length > 0) {
+        fluid.promise.sequence(promises).then(
+            function () { that.events.onSchemasUpdated.fire(that); },
+            function (error) { fluid.fail(error.message || error); }
+        );
+    }
+};
+
+fluid.defaults("gpii.schema.parser", {
+    gradeNames:    ["fluid.modelComponent"],
+    requiredRegexp: "/required$",
+    parserOptions: {},
+    uriTemplate:   "%schemaPath/%schemaKey",
+    messages: {
+        required: "You must provide a value for this field."
+    },
+    model: {
+        schemas: {}
+    },
+    events: {
+        onSchemasUpdated: null
+    },
+    members: {
+        parser: null,
+        dereferencedSchemas: {},
+        isReady: false
+    },
+    invokers: {
+        dereferenceSchema: {
+            funcName: "gpii.schema.parser.dereferenceSchema",
+            args: ["{that}", "{that}.options.schemaPath", "{arguments}.0"] // schemaPath, callback
+        },
+        resolveJsonPointer: {
+            funcName: "gpii.schema.parser.resolveJsonPointer",
+            args:     ["{that}", "{arguments}.0", "{arguments}.1"] // schemaKey, jsonPointer
+        },
+        evolveError: {
+            funcName: "gpii.schema.parser.evolveError",
+            args:     ["{that}", "{arguments}.0", "{arguments}.1"] // schemaKey, jsonPointer
+        }
+    },
+    listeners: {
+        "onSchemasUpdated.indicateReadiness": {
```

Comment:
Much of the complexity is added by the "dereferencer", which expects to be able to retrieve remote schemas.  I plan to resolve this by moving the dereferencer to the "inline schema" router, which will deliver schema content that has already been dereferenced.

With that moved, the validator will be ready to respond once it has retrieved its schema content, and firing a single `onSchemasLoaded` event will suffice.